Using vitest brings in @types/node in a basic setup

### Describe the bug

All in the title. Repro is the starter Vite project with Vitest installed.

### Reproduction

https://github.com/Faithfinder/vitest-types-node/blob/main/src/App.test.ts

Take a look in the linked file. "Buffer" looks ok to use.
Comment out the vitest import.
Now Buffer is unavailable.

This is true for the whole project (e.g, Buffer could be in any of the files).

### System Info

```shell
System:
    OS: Linux 5.15 Ubuntu 20.04.5 LTS (Focal Fossa)
    CPU: (8) x64 Intel(R) Core(TM) i7-9700K CPU @ 3.60GHz
    Memory: 1.05 GB / 5.79 GB
    Container: Yes
    Shell: 5.0.17 - /usr/bin/bash
  Binaries:
    Node: 18.14.0 - ~/.local/share/pnpm/node
    Yarn: 1.22.5 - /usr/bin/yarn
    npm: 9.3.1 - ~/.local/share/pnpm/npm
  npmPackages:
    @vitejs/plugin-react-swc: ^3.0.0 => 3.2.0 
    vite: ^4.1.0 => 4.1.4 
    vitest: ^0.28.5 => 0.28.5
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

How can I work around this issue? 🙄 
```
node_modules/@types/node/globals.d.ts:72:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'AbortSignal' must be of type '{ new (): AbortSignal; prototype: AbortSignal; abort(reason?: any): AbortSignal; timeout(milliseconds: number): AbortSignal; }', but here has type '{ new (): AbortSignal; prototype: AbortSignal; }'.

72 declare var AbortSignal: {
               ~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2090:13
    2090 declare var AbortSignal: {
                     ~~~~~~~~~~~
    'AbortSignal' was also declared here.
```

--- a/packages/vitest/package.json
+++ b/packages/vitest/package.json
@@ -146,6 +146,8 @@
     }
   },
   "dependencies": {
+    "@types/chai": "^4.3.5",
+    "@types/chai-subset": "^1.3.3",
     "@vitest/expect": "workspace:*",
     "@vitest/runner": "workspace:*",
     "@vitest/snapshot": "workspace:*",

--- a/packages/vitest/package.json
+++ b/packages/vitest/package.json
@@ -106,6 +106,7 @@
     "prepublishOnly": "pnpm build"
   },
   "peerDependencies": {
+    "@types/node": ">= 14",
     "@edge-runtime/vm": "*",
     "@vitest/browser": "*",
     "@vitest/ui": "*",
@@ -116,6 +117,9 @@
     "webdriverio": "*"
   },
   "peerDependenciesMeta": {
+    "@types/node": {
+      "optional": true
+    },
     "@vitest/ui": {
       "optional": true
     },
@@ -142,9 +146,6 @@
     }
   },
   "dependencies": {
-    "@types/chai": "^4.3.5",
-    "@types/chai-subset": "^1.3.3",
-    "@types/node": "*",
     "@vitest/expect": "workspace:*",
     "@vitest/runner": "workspace:*",
     "@vitest/snapshot": "workspace:*",

